src/cmd/linuxkit/run_vcenter.go

```diff
@@ -248,7 +260,7 @@ func addVMDK(ctx context.Context, vm *object.VirtualMachine, dss *object.Datasto
 		log.Fatalf("Unable to find SCSI device from VM configuration\n%v", err)
 	}
 	// The default is to have all persistent disks named linuxkit.vmdk
-	disk := devices.CreateDisk(controller, dss.Reference(), dss.Path(fmt.Sprintf("%s/%s", *newVM.vmName, "linuxkit.vmdk")))
+	disk := devices.CreateDisk(controller, dss.Reference(), dss.Path(fmt.Sprintf("%s/%s", *newVM.vmFolder, "linuxkit.vmdk")))
```

Comment:
It was to begin with, but then it became apparent that uploading  a`vmdk` from workstation to server required a lot more steps (image conversion from thin to fat, meant uploading GBs of zeros etc.) As there is a 1:1 mapping inside of VMware vCenter VMs I opted to keep it simple and in the event the `run` requires a persistent disk they just get a new disk inside of their VM folder to use. 

Happy to revisit this, it felt that uploading large existing disks or having to manipulate the disk controllers to support additional SCSI disks might be outside of what `run` should be doing :)